src/unit_tests/analysisd/test_decoder_syscollector.c

```diff
@@ -1726,153 +3321,154 @@ void test_syscollector_dbsync_hardware_valid_msg_inserted(void **state)
+    expect_any(__wrap_wdbc_query_ex, len);
+    will_return(__wrap_wdbc_query_ex, result_not_ok);
+    will_return(__wrap_wdbc_query_ex, 0);
+    expect_string(__wrap__mdebug1, formatted_msg, 
+            "Unable to send packages information to Wazuh DB.");
+
+    ret = DecodeSyscollector(lf, &sock);
+    assert_int_not_equal(ret, -1);
 }
 
-void test_syscollector_dbsync_os_valid_msg_with_number_pk(void **state)
+void test_syscollector_package_valid_without_ID (void **state)
 {
     Eventinfo *lf = *state;
 
-    const char *query = "agent 001 dbsync osinfo MODIFIED "
-        "{"
-            "\"scan_time\":\"2021/10/29 14:26:24\","
-            "\"hostname\":\"93\","
-            "\"architecture\":94,"
-            "\"os_name\":\"95\","
-            "\"os_version\":\"96\","
-            "\"os_codename\":\"97\","
-            "\"os_major\":\"98\","
-            "\"os_minor\":\"99\","
-            "\"os_patch\":\"100\","
-            "\"os_build\":\"101\","
-            "\"os_platform\":\"102\","
-            "\"sysname\":\"103\","
-            "\"release\":\"104\","
-            "\"version\":\"105\","
-            "\"os_release\":\"106\","
-            "\"checksum\":\"107\","
-            "\"os_display_version\":\"108\","
-            "\"reference\":\"110\""
-        "}";
+    expect_string(__wrap__mdebug1, formatted_msg, 
+            "Unable to send packages information to Wazuh DB.");
+
+    int ret = DecodeSyscollector(lf, NULL);
+
+    assert_int_not_equal(ret, -1);
+}
+
+void test_syscollector_package_program_end (void **state)
+{
+    Eventinfo *lf = *state;
+    const char *query = "agent 001 package del 100";
     const char *result = "ok ";
+    const char *result_not_ok = "not ok ";
     int sock = 1;
 
+    error_package = 0;
     expect_any(__wrap_wdbc_query_ex, *sock);
     expect_string(__wrap_wdbc_query_ex, query, query);
     expect_any(__wrap_wdbc_query_ex, len);
     will_return(__wrap_wdbc_query_ex, result);
     will_return(__wrap_wdbc_query_ex, 0);
 
     int ret = DecodeSyscollector(lf, &sock);
+    assert_int_not_equal(ret, 0);
+
+    // Invalid query out
+    expect_any(__wrap_wdbc_query_ex, *sock);
+    expect_string(__wrap_wdbc_query_ex, query, query);
+    expect_any(__wrap_wdbc_query_ex, len);
+    will_return(__wrap_wdbc_query_ex, result);
+    will_return(__wrap_wdbc_query_ex, -1);
+    expect_string(__wrap__mdebug1, formatted_msg, 
+            "Unable to send packages information to Wazuh DB.");
+
+    ret = DecodeSyscollector(lf, &sock);
+    assert_int_not_equal(ret, -1);
 
+    // Return ok after query error
+    ret = DecodeSyscollector(lf, &sock);
     assert_int_not_equal(ret, 0);
-}
 
-void test_syscollector_dbsync_with_no_type(void **state)
-{
-    Eventinfo *lf = *state;
-    int sock = 1;
-    const char type[] = {"dbsync"};
-    char error_log[128];
-    sprintf(error_log, INVALID_PREFIX, type);
+    // Return ok after query error, changing package_id
```

Comment:
We might set and clear the global variables with setup and teardown methods to avoid silently affecting other tests.

Reply:
Due to the number of functions defined, mainly because of the code coverage, and in the absence of other tools, it seems more appropriate to not continue adding functions excessively.